prowler/providers/aws/services/sagemaker/sagemaker_service.py

```diff
@@ -190,6 +192,26 @@ def __describe_training_job__(self, regional_clients):
                 f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
 
+    def __list_tags_for_resource__(self):
+        logger.info("SageMaker - List Tags...")
+        try:
+            for model in self.sagemaker_models:
+                regional_client = self.regional_clients[model.region]
+                response = regional_client.list_tags(ResourceArn=model.arn)["Tags"]
+                model.tags = response
+            for instance in self.sagemaker_notebook_instances:
+                regional_client = self.regional_clients[instance.region]
+                response = regional_client.list_tags(ResourceArn=instance.arn)["Tags"]
+                instance.tags = response
+            for job in self.sagemaker_training_jobs:
+                regional_client = self.regional_clients[job.region]
+                response = regional_client.list_tags(ResourceArn=job.arn)["Tags"]
+                job.tags = response
```

Comment:
I would split this function into three, to check each resource tags in a different logic.

Reply:
Done